### PR TITLE
new: add support to sqlcon to handle postgreSQL serialization_failure error

### PR DESCRIPTION
**Pull request by aaslamin**
## Related issue

- https://github.com/ory/hydra/issues/1719
- https://github.com/ory/hydra/issues/1735 

## Proposed changes

Add new error handling for the serialization error returned by postgreSQL when using the `REPEATABLE READ` isolation level. This error is returned when an UPDATE/DELETE (TXN2) statement was queued behind another UPDATE/DELETE (TXN1) statement on the same row. Once TXN1 completes and commits, TXN2 will fail to ensure consistency.

- postgreSQL error code appendix: https://www.postgresql.org/docs/9.6/errcodes-appendix.html
- error code: 40001

## Dependent PR(s) ⚠️ 

This error will be consumed by https://github.com/ory/hydra/pull/1766 so it can then return a `fosite. ErrSerializationFailure` which will then be [consumed by `fosite` to render an appropriate RFC 6749 error](https://github.com/ory/fosite/pull/402).

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
- [ ] I have read the [security policy](../security/policy)
- [x] I confirm that this pull request does not address a security vulnerability. If this pull request addresses a security
vulnerability, I confirm that I got green light (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation within the code base (if appropriate)

### CODE REVIEW

**Comment by aaslamin** [comment]
@aeneasr I noticed there is no test coverage here, I could add some if you like? ✨

**Comment by aaslamin** [comment]
How do I get the Semantic PR build check to trigger again? I modified my commit message to follow the style and force pushed another commit, but no luck 😢

File: sqlcon/error.go [hunk 1]
```diff
@@ -43,6 +47,8 @@ func HandleError(err error) error {
 		switch err.Code {
 		case "23505": // "unique_violation"
 			return errors.Wrap(ErrUniqueViolation, err.Error())
+		case "40001": // "serialization_failure"
```
Comment (aeneasr) [inline sqlcon/error.go]: Do we only care about PostgreSQL for now?

**Comment by aeneasr** [comment]
Released as v0.0.108

**Comment by aaslamin** [comment]
> LGTM! Should we add MySQL in another PR?

Yes. And also for cockroachDB if it is returning an error type different from what the postgreSQL driver is returning.

> Regarding tests: I'm not sure how to test the HandleError except for mocking the error which isn't very useful IMO. What do you think?

Yeah, that's what I was thinking. Value add is just to catch regressions due to things like accidental refactoring, etc.

> In any case, I'll merge this so you can continue working on the other PR!

Thank you! 🙌

File: sqlcon/error.go (hunk 1 above)
Comment (aaslamin) [reply sqlcon/error.go]: See https://github.com/ory/x/pull/129#issuecomment-603427081

**Pull request merged**